# Analysis of Fortune 500 Companies

### Import Libraries

In [0]:
import pandas as pd
import numpy as np

### Read in Dataset

In [0]:
#f500 = pd.read_csv('~/Google Drive/Colab Notebooks/data-science-portfolio-datasets/Fortune 500/f500.csv', index_col=0)
#f500.index.name = None

In [3]:
from google.colab import drive
drive.mount('/content/drive')
f500_file = open('/content/drive/My Drive/Colab Notebooks/data-science-portfolio-datasets/Fortune 500/f500.csv')
f500 = pd.read_csv(f500_file, index_col=0)

Go to this URL in a browser: https://accounts.google.com/o/oauth2/auth?client_id=947318989803-6bn6qk8qdgf4n4g3pfee6491hc0brc4i.apps.googleusercontent.com&redirect_uri=urn%3aietf%3awg%3aoauth%3a2.0%3aoob&response_type=code&scope=email%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdocs.test%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive%20https%3a%2f%2fwww.googleapis.com%2fauth%2fdrive.photos.readonly%20https%3a%2f%2fwww.googleapis.com%2fauth%2fpeopleapi.readonly

Enter your authorization code:
··········
Mounted at /content/drive


### Review Initial Info About Dataset

In [4]:
# view first 5 rows
f500.head()

rank  revenues  ...  employees  total_stockholder_equity
company                                   ...                                     
Walmart                      1    485873  ...    2300000                     77798
State Grid                   2    315199  ...     926067                    209456
Sinopec Group                3    267518  ...     713288                    106523
China National Petroleum     4    262573  ...    1512048                    301893
Toyota Motor                 5    254694  ...     364445                    157210

[5 rows x 16 columns]

In [5]:
# view last 5 rows
f500.tail()

rank  ...  total_stockholder_equity
company                               ...                          
Teva Pharmaceutical Industries   496  ...                     33337
New China Life Insurance         497  ...                      8507
Wm. Morrison Supermarkets        498  ...                      5111
TUI                              499  ...                      3006
AutoNation                       500  ...                      2310

[5 rows x 16 columns]

In [6]:
# view data types for each column
f500.info()

<class 'pandas.core.frame.DataFrame'>
Index: 500 entries, Walmart to AutoNation
Data columns (total 16 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   rank                      500 non-null    int64  
 1   revenues                  500 non-null    int64  
 2   revenue_change            498 non-null    float64
 3   profits                   499 non-null    float64
 4   assets                    500 non-null    int64  
 5   profit_change             436 non-null    float64
 6   ceo                       500 non-null    object 
 7   industry                  500 non-null    object 
 8   sector                    500 non-null    object 
 9   previous_rank             500 non-null    int64  
 10  country                   500 non-null    object 
 11  hq_location               500 non-null    object 
 12  website                   500 non-null    object 
 13  years_on_global_500_list  500 non-null    int64  
 14  em

### Create Sample Variable for One Column

In [7]:
industries = f500['industry']
industries

company
Walmart                                     General Merchandisers
State Grid                                              Utilities
Sinopec Group                                  Petroleum Refining
China National Petroleum                       Petroleum Refining
Toyota Motor                             Motor Vehicles and Parts
                                               ...               
Teva Pharmaceutical Industries                    Pharmaceuticals
New China Life Insurance          Insurance: Life, Health (stock)
Wm. Morrison Supermarkets                    Food and Drug Stores
TUI                                               Travel Services
AutoNation                                    Specialty Retailers
Name: industry, Length: 500, dtype: object

In [8]:
# determine data type for above variable
industries_type = type(industries)
industries_type

pandas.core.series.Series

The above variable is a 'series' data type, Pandas' implementation of a one-dimensional array. Alternatively, Pandas uses dataframes for two-dimensional arrays.

### Select Multiple Columns

In [9]:
# select one column
countries = f500['country']
countries

company
Walmart                               USA
State Grid                          China
Sinopec Group                       China
China National Petroleum            China
Toyota Motor                        Japan
                                   ...   
Teva Pharmaceutical Industries     Israel
New China Life Insurance            China
Wm. Morrison Supermarkets         Britain
TUI                               Germany
AutoNation                            USA
Name: country, Length: 500, dtype: object

In [10]:
# select multiple columns based on distinct column names
revenues_years = f500[['revenues','years_on_global_500_list']]
revenues_years

revenues  years_on_global_500_list
company                                                           
Walmart                           485873                        23
State Grid                        315199                        17
Sinopec Group                     267518                        19
China National Petroleum          262573                        17
Toyota Motor                      254694                        23
...                                  ...                       ...
Teva Pharmaceutical Industries     21903                         1
New China Life Insurance           21796                         2
Wm. Morrison Supermarkets          21741                        13
TUI                                21655                        23
AutoNation                         21609                        12

[500 rows x 2 columns]

In [11]:
# select multiple columns using slicing
ceo_to_sector = f500.loc[:,'ceo':'sector']
ceo_to_sector

ceo  ...                  sector
company                                              ...                        
Walmart                         C. Douglas McMillon  ...               Retailing
State Grid                                  Kou Wei  ...                  Energy
Sinopec Group                             Wang Yupu  ...                  Energy
China National Petroleum              Zhang Jianhua  ...                  Energy
Toyota Motor                            Akio Toyoda  ...  Motor Vehicles & Parts
...                                             ...  ...                     ...
Teva Pharmaceutical Industries    Yitzhak Peterburg  ...             Health Care
New China Life Insurance                   Wan Feng  ...              Financials
Wm. Morrison Supermarkets            David T. Potts  ...      Food & Drug Stores
TUI                               Friedrich Joussen  ...       Business Services
AutoNation                       Michael J. Jackson  ...               Retailing

[500 rows x 3 columns]

### Select Multiple Rows

In [12]:
# select data from a single row - results in a series object
toyota = f500.loc['Toyota Motor']
toyota

rank                                                   5
revenues                                          254694
revenue_change                                       7.7
profits                                          16899.3
assets                                            437575
profit_change                                      -12.3
ceo                                          Akio Toyoda
industry                        Motor Vehicles and Parts
sector                            Motor Vehicles & Parts
previous_rank                                          8
country                                            Japan
hq_location                                Toyota, Japan
website                     http://www.toyota-global.com
years_on_global_500_list                              23
employees                                         364445
total_stockholder_equity                          157210
Name: Toyota Motor, dtype: object

In [13]:
# select data from multiple named rows
drink_companies = f500.loc[['Anheuser-Busch InBev', 'Coca-Cola', 'Heineken Holding']]
drink_companies

rank  revenues  ...  employees  total_stockholder_equity
company                               ...                                     
Anheuser-Busch InBev   206     45905  ...     206633                     71339
Coca-Cola              235     41863  ...     100300                     23062
Heineken Holding       468     23044  ...      73525                      6958

[3 rows x 16 columns]

In [14]:
# select data from specific rows and columms using slicing
middle_companies = f500.loc['Tata Motors':'Nationwide','rank':'country']
middle_companies

rank  revenues  ...  previous_rank  country
company                                          ...                        
Tata Motors                       247     40329  ...            226    India
Aluminum Corp. of China           248     40278  ...            262    China
Mitsui                            249     40275  ...            245    Japan
Manulife Financial                250     40238  ...            394   Canada
China Minsheng Banking            251     40234  ...            221    China
China Pacific Insurance (Group)   252     40193  ...            251    China
American Airlines Group           253     40180  ...            236      USA
Nationwide                        254     40074  ...            241      USA

[8 rows x 11 columns]

### Series Methods

In [15]:
# use value_counts() method to determine number of distinct values
countries = f500['country']
country_counts = countries.value_counts()
country_counts

USA             132
China           109
Japan            51
France           29
Germany          29
Britain          24
South Korea      15
Netherlands      14
Switzerland      14
Canada           11
Spain             9
Italy             7
Australia         7
India             7
Brazil            7
Taiwan            6
Russia            4
Ireland           4
Singapore         3
Sweden            3
Mexico            2
Luxembourg        1
Turkey            1
Denmark           1
U.A.E             1
Finland           1
Thailand          1
Belgium           1
Norway            1
Saudi Arabia      1
Israel            1
Indonesia         1
Malaysia          1
Venezuela         1
Name: country, dtype: int64

In [16]:
# select number of companies from India
india = country_counts.loc['India']
india

7

In [17]:
# select number of companies from North American countries
north_america = country_counts.loc[['USA', 'Canada', 'Mexico']]
north_america

USA       132
Canada     11
Mexico      2
Name: country, dtype: int64

In [18]:
# select specific rows and columns by name
big_movers = f500.loc[['Aviva', 'HP', 'JD.com', 'BHP Billiton'],['rank', 'previous_rank']]
big_movers

rank  previous_rank
company                          
Aviva           90            279
HP             194             48
JD.com         261            366
BHP Billiton   350            168

In [19]:
# select range of rows by slicing and specific columns by name
bottom_companies = f500.loc['National Grid':'AutoNation', ['rank', 'sector', 'country']]
bottom_companies

rank              sector  country
company                                                                 
National Grid                           491              Energy  Britain
Dollar General                          492           Retailing      USA
Telecom Italia                          493  Telecommunications    Italy
Xiamen ITG Holding Group                494         Wholesalers    China
Xinjiang Guanghui Industry Investment   495         Wholesalers    China
Teva Pharmaceutical Industries          496         Health Care   Israel
New China Life Insurance                497          Financials    China
Wm. Morrison Supermarkets               498  Food & Drug Stores  Britain
TUI                                     499   Business Services  Germany
AutoNation                              500           Retailing      USA

### Vectorized Operations

#### Use vectorized operations to perform calculations on pandas series

Calculate the current year vs. previous year change in rank for each company, and determine the biggest increase / biggest decrease in rank.

Below, the calculated biggest decrease in rank shown was -500. This is not possible, as the maximum decrease would be -499 if a company dropped from 1st place to 500th place.


In [20]:
# calculate previous year to current year change in rank
rank_change = f500['previous_rank'] - f500['rank']
rank_change

company
Walmart                             0
State Grid                          0
Sinopec Group                       1
China National Petroleum           -1
Toyota Motor                        3
                                 ... 
Teva Pharmaceutical Industries   -496
New China Life Insurance          -70
Wm. Morrison Supermarkets         -61
TUI                               -32
AutoNation                       -500
Length: 500, dtype: int64

In [21]:
# find the company with the biggest increase in rank
rank_change_max = rank_change.max()
rank_change_max

226

In [22]:
# find the company with the biggest descrease in rank
rank_change_min = rank_change.min()
rank_change_min

-500

To identify the cause of the -500 data inconsistency, the code below displays common statistics for both the **rank** and **previous_rank** columns. The **previous_rank** column includes a minimum value of 0, likely indicating that the respective companies were not on the top 500 list during the previous year.

33 companies contains the 0 value in the previous_rank column.

In [23]:
# calculate stats for the rank column
rank = f500['rank']
rank_desc = rank.describe()
rank_desc

count    500.000000
mean     250.500000
std      144.481833
min        1.000000
25%      125.750000
50%      250.500000
75%      375.250000
max      500.000000
Name: rank, dtype: float64

In [24]:
# calculate the stats for the previous_rank column
prev_rank = f500['previous_rank']
prev_rank_desc = prev_rank.describe()
prev_rank_desc

count    500.000000
mean     222.134000
std      146.941961
min        0.000000
25%       92.750000
50%      219.500000
75%      347.250000
max      500.000000
Name: previous_rank, dtype: float64

In [25]:
# calculate the number of countries with a zero value in the Previous Rank column
zero_previous_rank = f500['previous_rank'].value_counts().loc[0]
zero_previous_rank

33

In [26]:
# calculate the maximum value for each of the numeric columns
max_f500 = f500.max(numeric_only=True)
max_f500

rank                            500.0
revenues                     485873.0
revenue_change                  442.3
profits                       45687.0
assets                      3473238.0
profit_change                  8909.5
previous_rank                   500.0
years_on_global_500_list         23.0
employees                   2300000.0
total_stockholder_equity     301893.0
dtype: float64

In [27]:
# calculate common statistics for the numeric columns
f500_desc = f500.describe()
f500_desc

rank       revenues  ...     employees  total_stockholder_equity
count  500.000000     500.000000  ...  5.000000e+02                500.000000
mean   250.500000   55416.358000  ...  1.339983e+05              30628.076000
std    144.481833   45725.478963  ...  1.700878e+05              43642.576833
min      1.000000   21609.000000  ...  3.280000e+02             -59909.000000
25%    125.750000   29003.000000  ...  4.293250e+04               7553.750000
50%    250.500000   40236.000000  ...  9.291050e+04              15809.500000
75%    375.250000   63926.750000  ...  1.689172e+05              37828.500000
max    500.000000  485873.000000  ...  2.300000e+06             301893.000000

[8 rows x 10 columns]

### Boolean Indexing / Data Replacement

The CEO of the Dow Chemical company has changed. Using boolean indexing, change the value in this location.

In [28]:
# identify the current Dow Chemical CEO
f500.loc['Dow Chemical','ceo']
print('Old CEO: ' + f500.loc['Dow Chemical', 'ceo'])

# replace with new CEO's name
f500.loc['Dow Chemical', 'ceo'] = 'Jim Fitterling'
print('New CEO: ' + f500.loc['Dow Chemical', 'ceo'])

Old CEO: Andrew N. Liveris
New CEO: Jim Fitterling


Create a new dataframe that only includes companies in the "Motor Vehicles and Parts" industry

In [29]:
motor_bool = f500['industry'] == 'Motor Vehicles and Parts'
motor_countries = f500.loc[motor_bool, 'country']
motor_countries

company
Toyota Motor                                 Japan
Volkswagen                                 Germany
Daimler                                    Germany
General Motors                                 USA
Ford Motor                                     USA
Honda Motor                                  Japan
SAIC Motor                                   China
Nissan Motor                                 Japan
BMW Group                                  Germany
Dongfeng Motor                               China
Robert Bosch                               Germany
Hyundai Motor                          South Korea
China FAW Group                              China
Beijing Automotive Group                     China
Peugeot                                     France
Renault                                     France
Kia Motors                             South Korea
Continental                                Germany
Denso                                        Japan
Guangzhou Automobile In

Identify all rows where the previous_rank value is zero. As this value actually represents that the respective company was not on the Fortune 500 list during the previous year, replace these values with NumPy's nan (i.e. NULL).

In [30]:
prev_rank_before = f500["previous_rank"].value_counts(dropna=False).head()
prev_rank_before

0      33
159     1
147     1
148     1
149     1
Name: previous_rank, dtype: int64

In [0]:
# replace previous_rank 0 values with NULL
f500.loc[f500['previous_rank'] == 0, 'previous_rank'] = np.nan

In [32]:
prev_rank_after = f500['previous_rank'].value_counts(dropna=False).head()
prev_rank_after

NaN      33
471.0     1
234.0     1
125.0     1
166.0     1
Name: previous_rank, dtype: int64

Add a new column that calculates the change in value between Rank and Previous Rank columns.

In [33]:
# create column calculating change in rank
f500['rank_change'] = f500['previous_rank'] - f500['rank']

# output stats for new rank change column
rank_change_desc = f500['rank_change'].describe()
rank_change_desc

count    467.000000
mean      -3.533191
std       44.293603
min     -199.000000
25%      -21.000000
50%       -2.000000
75%       10.000000
max      226.000000
Name: rank_change, dtype: float64

Identify the two most common industries for companies headquartered in the USA

In [34]:
# create new df with only American companies
company_usa = f500.loc[f500['country'] == 'USA']

# create series of two most common industries
industry_usa = company_usa['industry'].value_counts().head(2)                     
industry_usa

Banks: Commercial and Savings               8
Insurance: Property and Casualty (Stock)    7
Name: industry, dtype: int64

Identify the threemost common sectors for companies headquartered in China

In [35]:
# create new df with only Chinese companies
company_china = f500.loc[f500['country'] == 'China']

# create series of two most common industries
sector_china = company_china['sector'].value_counts().head(3)                     
sector_china

Financials     25
Energy         22
Wholesalers     9
Name: sector, dtype: int64

### Pandas read_csv() Info

In [36]:
# select the rank, revenues, and revenue_change columns from the first five rows
f500_selection = f500[['rank','revenues','revenue_change']].head()
f500_selection

rank  revenues  revenue_change
company                                                 
Walmart                      1    485873             0.8
State Grid                   2    315199            -4.4
Sinopec Group                3    267518            -9.1
China National Petroleum     4    262573           -12.3
Toyota Motor                 5    254694             7.7

Up until this point, we have used the actual names of the companies as the index for each row. Now, we'll use a sequential integer as the index. This is the more common way of using Pandas dataframes.

In [37]:
# read f500.csv again, but do not assign company name as index

#f500 = pd.read_csv('~/Google Drive/Colab Notebooks/data-science-portfolio-datasets/Fortune 500/f500.csv')

f500_file = open('/content/drive/My Drive/Colab Notebooks/data-science-portfolio-datasets/Fortune 500/f500.csv')
f500 = pd.read_csv(f500_file)


f500.loc[f500["previous_rank"] == 0, "previous_rank"] = np.nan

f500.head()

company  rank  ...  employees  total_stockholder_equity
0                   Walmart     1  ...    2300000                     77798
1                State Grid     2  ...     926067                    209456
2             Sinopec Group     3  ...     713288                    106523
3  China National Petroleum     4  ...    1512048                    301893
4              Toyota Motor     5  ...     364445                    157210

[5 rows x 17 columns]

### iloc()

Use Pandas iloc function to select specific rows / columns based on the integer placement in the dataframe.

In [38]:
# select the entire 5th row of the dataframe
fifth_row = f500.iloc[4]
fifth_row

company                                     Toyota Motor
rank                                                   5
revenues                                          254694
revenue_change                                       7.7
profits                                          16899.3
assets                                            437575
profit_change                                      -12.3
ceo                                          Akio Toyoda
industry                        Motor Vehicles and Parts
sector                            Motor Vehicles & Parts
previous_rank                                          8
country                                            Japan
hq_location                                Toyota, Japan
website                     http://www.toyota-global.com
years_on_global_500_list                              23
employees                                         364445
total_stockholder_equity                          157210
Name: 4, dtype: object

In [39]:
# select the value in the first row of the Company column
company_value = f500.iloc[0,0]

company_value

'Walmart'

In [40]:
# select the first three rows
first_three_rows = f500.iloc[:3]
first_three_rows

company  rank  ...  employees  total_stockholder_equity
0        Walmart     1  ...    2300000                     77798
1     State Grid     2  ...     926067                    209456
2  Sinopec Group     3  ...     713288                    106523

[3 rows x 17 columns]

In [41]:
# select 1st and 7th rows and 1st five columns
first_seventh_row_slice = f500.iloc[[0,6],:5]
first_seventh_row_slice

company  rank  revenues  revenue_change  profits
0            Walmart     1    485873             0.8  13643.0
6  Royal Dutch Shell     7    240033           -11.8   4575.0

### Pandas Boolean Operators

In [42]:
# select all rows from f500 that have a null value for the previous_rank column. Select only the company, rank, and previous_rank columns.
null_previous_rank = f500.loc[f500['previous_rank'].isnull(),['company','rank','previous_rank']]
null_previous_rank

company  rank  previous_rank
48                   Legal & General Group    49            NaN
90                                  Uniper    91            NaN
123                      Dell Technologies   124            NaN
138                 Anbang Insurance Group   139            NaN
140                        Albertsons Cos.   141            NaN
180             Hewlett Packard Enterprise   181            NaN
267                           Hengli Group   268            NaN
271         Johnson Controls International   272            NaN
341                                  Chubb   342            NaN
375                 Charter Communications   376            NaN
376                              Medtronic   377            NaN
392                               Facebook   393            NaN
409                            Kraft Heinz   410            NaN
414                                  Nokia   415            NaN
431                          Standard Life   432            NaN
440            Brookfield Asset Management   441            NaN
441        Mercantil Servicios Financieros   442            NaN
446                            Samsung C&T   447            NaN
456                        Heraeus Holding   457            NaN
458                Yango Financial Holding   459            NaN
461                  Alibaba Group Holding   462            NaN
463                            Progressive   464            NaN
466                Country Garden Holdings   467            NaN
472                                 Altice   473            NaN
477                       Tencent Holdings   478            NaN
482                                  Aflac   483            NaN
484                  Suning Commerce Group   485            NaN
487                             Xiamen C&D   488            NaN
491                         Dollar General   492            NaN
493               Xiamen ITG Holding Group   494            NaN
494  Xinjiang Guanghui Industry Investment   495            NaN
495         Teva Pharmaceutical Industries   496            NaN
499                             AutoNation   500            NaN

In [43]:
top5_null_prev_rank = null_previous_rank.iloc[:5]
top5_null_prev_rank

company  rank  previous_rank
48    Legal & General Group    49            NaN
90                   Uniper    91            NaN
123       Dell Technologies   124            NaN
138  Anbang Insurance Group   139            NaN
140         Albertsons Cos.   141            NaN

In [44]:
previously_ranked = f500[f500['previous_rank'].notnull()]

rank_change = previously_ranked['previous_rank'] - previously_ranked['rank']

f500['rank_change'] = rank_change
f500

company  ...  rank_change
0                           Walmart  ...          0.0
1                        State Grid  ...          0.0
2                     Sinopec Group  ...          1.0
3          China National Petroleum  ...         -1.0
4                      Toyota Motor  ...          3.0
..                              ...  ...          ...
495  Teva Pharmaceutical Industries  ...          NaN
496        New China Life Insurance  ...        -70.0
497       Wm. Morrison Supermarkets  ...        -61.0
498                             TUI  ...        -32.0
499                      AutoNation  ...          NaN

[500 rows x 18 columns]

Select all companies with revenues over 100 billion and negative profits from the f500 dataframe. The result should include all columns.

In [47]:
# Create a boolean array that selects the companies with revenues greater than 100 billion.
large_revenue = f500['revenues'] > 100000

# Create a boolean array that selects the companies with profits less than 0.
negative_profits = f500['profits'] < 0

# Combine large_revenue and negative_profits. 
combined = large_revenue & negative_profits

# Use combined to filter f500.
big_rev_neg_profit = f500.loc[combined]
big_rev_neg_profit

company  rank  ...  total_stockholder_equity  rank_change
32  Japan Post Holdings    33  ...                     91532          4.0
44              Chevron    45  ...                    145556        -14.0

[2 rows x 18 columns]

Select all rows for companies headquartered in either Brazil or Venezuela. 

In [49]:
brazil_venezuela = f500[(f500['country'] == 'Brazil') | (f500['country'] == 'Venezuela')]
brazil_venezuela

company  ...  rank_change
74                         Petrobras  ...        -17.0
112            Itau Unibanco Holding  ...         46.0
150                  Banco do Brasil  ...        -36.0
153                   Banco Bradesco  ...         55.0
190                              JBS  ...         -6.0
369                             Vale  ...         47.0
441  Mercantil Servicios Financieros  ...          NaN
486                Ultrapar Holdings  ...        -13.0

[8 rows x 18 columns]

Select the first five companies in the Technology sector that are not headquartered in the USA from the f500 dataframe. 

In [51]:
tech_outside_usa = f500[~(f500['country'] == 'USA') & (f500['sector'] == 'Technology')].head()
tech_outside_usa

company  rank  ...  total_stockholder_equity  rank_change
14           Samsung Electronics    15  ...                    154376         -2.0
26    Hon Hai Precision Industry    27  ...                     33476         -2.0
70                       Hitachi    71  ...                     26632          8.0
82   Huawei Investment & Holding    83  ...                     20159         46.0
104                         Sony   105  ...                     22415          8.0

[5 rows x 18 columns]

Find the company headquartered in Japan with the largest number of employees.

In [57]:
japan_companies = f500[f500['country'] == 'Japan']
japan_companies_sorted = japan_companies.sort_values('employees', ascending=False)
top_company = japan_companies_sorted.iloc[0]
top_japanese_employer = top_company['company']
top_japanese_employer

'Toyota Motor'

### Pandas Loops

In this exercise, we're going to produce a dictionary of the top employer in each country

In [72]:
#pd.set_option("display.max_rows", None, "display.max_columns", None)
top_employer_by_country = {}

countries = f500['country'].unique()

for country in countries:
  selected_countries = f500[f500['country'] == country]
  sorted_countries = selected_countries.sort_values('employees', ascending=False)
  first_company_row = sorted_countries.iloc[0]
  first_company_name = first_company_row['company']
  top_employer_by_country[country] = first_company_name

top_employer_by_country

{'Australia': 'Wesfarmers',
 'Belgium': 'Anheuser-Busch InBev',
 'Brazil': 'JBS',
 'Britain': 'Compass Group',
 'Canada': 'George Weston',
 'China': 'China National Petroleum',
 'Denmark': 'Maersk Group',
 'Finland': 'Nokia',
 'France': 'Sodexo',
 'Germany': 'Volkswagen',
 'India': 'State Bank of India',
 'Indonesia': 'Pertamina',
 'Ireland': 'Accenture',
 'Israel': 'Teva Pharmaceutical Industries',
 'Italy': 'Poste Italiane',
 'Japan': 'Toyota Motor',
 'Luxembourg': 'ArcelorMittal',
 'Malaysia': 'Petronas',
 'Mexico': 'America Movil',
 'Netherlands': 'EXOR Group',
 'Norway': 'Statoil',
 'Russia': 'Gazprom',
 'Saudi Arabia': 'SABIC',
 'Singapore': 'Flex',
 'South Korea': 'Samsung Electronics',
 'Spain': 'Banco Santander',
 'Sweden': 'H & M Hennes & Mauritz',
 'Switzerland': 'Nestle',
 'Taiwan': 'Hon Hai Precision Industry',
 'Thailand': 'PTT',
 'Turkey': 'Koc Holding',
 'U.A.E': 'Emirates Group',
 'USA': 'Walmart',
 'Venezuela': 'Mercantil Servicios Financieros'}

### Return on Assets

Create a new column roa in the f500 dataframe, containing the return on assets metric for each company.

In [81]:
roa = f500['profits'] / f500['assets']

f500['roa'] = roa

top_roa_by_sector = {}

sectors = f500['sector'].unique()

for sector in sectors:
  selected_companies = f500[f500['sector'] == sector]
  sorted_companies = selected_companies.sort_values('roa', ascending=False)
  top_company = sorted_companies.iloc[0]
  top_company_name = top_company['company']
  top_roa_by_sector[sector] = top_company_name

top_roa_by_sector

{'Aerospace & Defense': 'Lockheed Martin',
 'Apparel': 'Nike',
 'Business Services': 'Adecco Group',
 'Chemicals': 'LyondellBasell Industries',
 'Energy': 'National Grid',
 'Engineering & Construction': 'Pacific Construction Group',
 'Financials': 'Berkshire Hathaway',
 'Food & Drug Stores': 'Publix Super Markets',
 'Food, Beverages & Tobacco': 'Philip Morris International',
 'Health Care': 'Gilead Sciences',
 'Hotels, Restaurants & Leisure': 'McDonald’s',
 'Household Products': 'Unilever',
 'Industrials': '3M',
 'Materials': 'CRH',
 'Media': 'Disney',
 'Motor Vehicles & Parts': 'Subaru',
 'Retailing': 'H & M Hennes & Mauritz',
 'Technology': 'Accenture',
 'Telecommunications': 'KDDI',
 'Transportation': 'Delta Air Lines',
 'Wholesalers': 'McKesson'}